openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -75,48 +79,263 @@ export function closePop(){
     });
 }
 
-export default function init(){
-    var $searchResults = $('header#header-bar .search-component ul.search-results');
-    var $searchInput = $('header#header-bar .search-component .search-bar-input input[type="text"]');
-    var cover_url = function(id) {
-        return `//covers.openlibrary.org/b/id/${id}-S.jpg`
-    };
-    // stores the state of the search result for resizing window
-    var instantSearchResultState = false;
-    var searchModes, searchModeDefault, defaultFacet, searchFacets, composeSearchUrl, marshalBookSearchQuery, renderInstantSearchResults, setFacet, setMode, setSearchMode, options, q, parts, enteredSearchMinimized, searchExpansionActivated, toggleSearchbar, renderInstantSearchResult, val, facet_value;
-
-    // searches should be cancelled if you click anywhere in the page
-    $('body').on('click', function () {
-        instantSearchResultState = false;
-        $searchResults.empty();
-    });
-    // but clicking search input should not empty search results.
-    $searchInput.on('click', false);
+const SEARCH_MODES = ['everything', 'ebooks', 'printdisabled'];
+const SEARCH_MODE_DEFAULT = 'ebooks';
+const SEARCH_FACETS = {
+    'title': 'books',
+    'author': 'authors',
+    'lists': 'lists',
+    'subject': 'subjects',
+    'all': 'all',
+    'advanced': 'advancedsearch',
+    'text': 'inside'
+};
+const DEFAULT_SEARCH_FACET = 'all';
+const RENDER_INSTANT_SEARCH_RESULT = {
+    books(work) {
+        const author_name = work.author_name ? work.author_name[0] : '';
+        $('header#header-bar .search-component ul.search-results').append(
```

Comment:
Since these are template like, I'd like to see these in separate files so that when we have templating that works on server and client, we can merge these.

Reply:
Let me know if you still think this is an issue in the latest iteration.